### DISCUSSION TITLE
Specify websocket protocol for instances

### DISCUSSION THREAD

**Discussion by veN1337** [mainsail-crew/mainsail#1764] [Help] [answered]
Hi,  
I'm running Mainsail in a container and specify my remote Moonraker instance in the `config.json`.  
Because the Moonsail container is accessed through a reverse proxy with https, the websocket protocol for the Moonraker instance is automatically assumed as `wss://`. This is not corrected, because `wss://` is not enabled on Moonraker and the connection fails . Due to technical limitations I can not enable `wss://` on Moonraker and would like to continue using insecure websocket (`ws://`). If I turn off https on the reverse proxy, Moonsail assumes `ws` as protocol and the connection works again.  
Is there a way to manually specify the protocol for a Moonraker instance?

**Comment by meteyou** [comment, marked as answer]
No this is not allowed, because this is a browser limitation. So this function is also "hardcoded" in Mainsail. Every browser only allows to connect between the same "secure level" so http websites can only connect to ws sockets and https websites can only connect to wss sockets. So a cross connection is not allowed by the browser itself. So wenn we allow this in Mainsail, you will not be able to connect to your printer with a setup like this.

So you can only host your Mainsail instance without SSL (http) or add a cert to Moonraker and connect via wss.

**Comment by veN1337** [comment]
Oh ok, I wasn't aware of that browser limitation or design/security decision

Thanks for explaining :)
